lib/plugins/threaded-signing/plugin.c

```diff
@@ -184,6 +184,13 @@ signing_worker_thread(void *user_data)
       // report the error when getting the signature.
       self->output_buffer[self->output_buffer_idx].signing_error = (status != SV_OK);
 
+      // |output_buffer| is full. Buffers this long are not supported.
+      if (self->output_buffer_idx >= MAX_BUFFER_LENGTH) {
+        self->is_running = false;
+        sv_threaded_plugin_reset(self);
+        goto done;
```

Comment:
One could argue that only a failing memory allocation should stop the thread. What do you propose?

Reply:
I propose that we stop the thread if the buffer is full. Buffer size is large enough to handle delays, hence something bad has happened if the buffer is full and we should return some kind of error code, stop the thread and free memory.

Reply:
Ack